Add section for troubleshooting video delay (#123)
foxglove/docs: docs/3-visualization/4-panels/image.md

- Scroll – Zoom in and out

> ## Troubleshooting Video Delay

Reviewer: ```suggestion
## Troubleshooting video delay
```

> You may see errors for `foxglove.CompressedVideo` topics stating that the frame being displayed is delayed relative to the most recent frame. Video decoding is implemented differently on each platform, and some platforms may experience delays while others do not. It is possible to configure your video stream to optimize for low-latency decoding, but it may be impossible to _guarantee_ that a video stream can be decoded with zero latency on all platforms.

Reviewer 2: ```suggestion
You may see errors for `foxglove.CompressedVideo` topics stating that the frame being displayed is delayed relative to the most recent frame. Video decoding very by platform, and some platforms may experience delays while others do not. It is possible to configure your video stream to optimize for low-latency decoding, but it may be impossible to _guarantee_ that a video stream can be decoded with zero latency on all platforms.
```

Reviewer: I don’t think those words are as words as you might have intended

Author: how about:
```suggestion
You may see errors for `foxglove.CompressedVideo` topics stating that the frame being displayed is delayed relative to the most recent frame. Video decoding behavior can vary across platforms, and some platforms may experience delays while others do not. It is possible to configure your video stream to optimize for low-latency decoding, but it may be impossible to _guarantee_ that a video stream can be decoded with zero latency on all platforms.
```

Reviewer 2: 👍

> Here are some tips on how configure the video encoding to best achieve low-latency decoding:

Reviewer 2: ```suggestion
Some tips on how configure the video encoding to best achieve low-latency decoding:
```

> You may see delay errors for `foxglove.CompressedVideo` topics saying that the frame being displayed could be X number of frames behind the most recent frame. What this means is that the video decoder that's being needs to fill up an internal buffer before it starts returning decoded frames. This could due to the configuration of the video stream, but whether this behavior occurs is very inherent to the decoder on your platform and its capabilities. On some platforms achieving zero latency decoding could be impossible to guarantee, but there are ways to configure your video stream to give a best-effort attempt at achieving low-latency when decoding.

Reviewer: ```suggestion
You may see errors for `foxglove.CompressedVideo` topics stating that the frame being displayed is delayed relative to the most recent frame. Video decoding is implemented differently on each platform, and some platforms may experience delays while others do not. It is possible to configure your video stream to optimize for low-latency decoding, but it may be impossible to _guarantee_ that a video stream can be decoded with zero latency on all platforms.
```

>  - Encoder configuration could require an internal buffer for decoding. This _can_ happen if your stream uses a profile which supports B frames, even if you don't use them. We recommend the `BASELINE` profile for `h264` encoded streams because it does not support B frames.

Reviewer: ```suggestion
 - Use the `BASELINE` profile for `h264` encoded streams. This profile is preferred because it does not support B frames. Profiles which support B frames may introduce decoding delay, even if the encoded stream doesn't actually contain B frames.
```

 - Disable frame reordering on your encoder.

>  - Lower profile levels (`level_idc`) usually have smaller buffers. This would result in a lower-latency decoding.

Reviewer: ```suggestion
 - Lower profile levels (`level_idc`) usually require smaller buffers, resulting in lower-latency decoding.
```

 - If your encoder allows it, use a `bitstream_restriction` in the VUI parameters to limit the size of the buffer (`max_dec_frame_buffering`) and also disable frame reordering (`max_num_reorder_frames`).

System CPU load and power consumption may also contribute to decoding delays. Hardware-accelerated decoding is generally faster and more energy-efficient. You can check that your platform supports hardware-accelerated decoding by opening Google Chrome and entering `chrome://gpu` in the address bar. If you do not see `Video Decode: Hardware accelerated` on this page, but you believe that it should be supported, then you may need to take additional platform-specific steps to enable it.

> These steps should help your platform achieve low-latency decoding, but remember that guaranteeing it may not be possible depending on the platform and device being used.

Reviewer 2: ```suggestion
These steps can help your platform achieve low-latency decoding, but there is no guarantee depending on the platform and device being used.
```

> Another possibility why this might be happening is because of decoder responding to high-CPU load or energy considerations along with the use of software-accelerated decoding (CPU based). You might want to check that hardware decoding is active and/or supported on your platform. Hardware-accelerated decoding is generally faster, and more energy efficient. You can see whether it's disabled by going to `chrome://gpu` in your chrome address bar. If you do not see `Video Decode: Hardware accelerated`, then you will want to look into seeing if decoder hardware acceleration is supported for your platform, and if it is, take the platform-specific steps to enable it.

Reviewer: ```suggestion
System CPU load and power consumption may also contribute to decoding delays. Hardware-accelerated decoding is generally faster and more energy-efficient. You can check that your platform supports hardware-accelerated decoding by opening Google Chrome and entering `chrome://gpu` in the address bar. If you do not see `Video Decode: Hardware accelerated` on this page, but you believe that it should be supported, then you may need to take additional platform-specific steps to enable it.
```

> Again, due to the decoder behavior being inherent to your platform and device it is not a guarantee that decoding will be low latency, but these are things you can do to allow your platform to give a best effort at low-latency decoding.

Reviewer: ```suggestion
These steps should help your platform achieve low-latency decoding, but remember that guaranteeing it may not be possible depending on the platform and device being used.
```
